scripts/check-docs/src/lib.rs

```diff
@@ -191,21 +194,83 @@ pub fn extract_starts_and_ends(
     Ok((begins, ends))
 }
 
-pub fn search_for_patterns_in_project(pattern: &str) -> anyhow::Result<String> {
+pub fn parse_md_files(text_w_files: String, path: &str) -> HashSet<PathBuf> {
+    let apply_regex = |regex: Regex| {
+        text_w_files
+            .lines()
+            .filter_map(|line| regex.captures(line))
+            .map(|capture| {
+                PathBuf::from(path)
+                    .join(&capture[1])
+                    .canonicalize()
+                    .expect("could not canonicalize md path")
+            })
+            .collect()
+    };
+
+    apply_regex(Regex::new(r"\((.*.md)\)").expect("could not contstruct regex"))
```

Comment:
Note the escaped `\.md` change:
```suggestion
    let regex = Regex::new(r"\((.*\.md)\)").expect("could not construct regex");

    text_w_files
        .lines()
        .filter_map(|line| regex.captures(line))
        .map(|capture| {
            PathBuf::from(path)
                .join(&capture[1])
                .canonicalize()
                .expect("could not canonicalize md path")
        })
        .collect()
```